generator/main.ts

```diff
@@ -32,14 +32,36 @@ async function generateInterface(input_idl_path: string, output_path: string) {
     file.read(path.resolve(TEMPLATE_DIR, 'interface_header.njk')),
     file.read(path.resolve(TEMPLATE_DIR, 'interface_cpp.njk'))
   ]);
+
+  const [bacardi_cpp_tmpl, bacardi_header_tmpl] = await Promise.all([
+    file.read(path.resolve(TEMPLATE_DIR, 'bacardi_header.njk')),
+    file.read(path.resolve(TEMPLATE_DIR, 'bacardi_cpp.njk'))
+  ]);
+
   const idl_name = input_idl_path.replace('.idl', '').replace('.', '');
+
   const header_file_path = path.resolve(output_path, idl_name + '_bridge.h');
   const cpp_file_path = path.resolve(output_path, idl_name + '_bridge.cc');
 
+  const bacardi_cpp_path = path.resolve(
+      output_path,
+      path.dirname(idl_name) + '/' +
+          'bacardi.cc');
+  const bacardi_header_path = path.resolve(
+      output_path,
+      path.dirname(idl_name) + '/' +
+          'bacardi.h');
+
   return Promise.all([
     file.write(
         header_file_path, nunjucks.renderString(header_tmpl, idl_interface)),
-    file.write(cpp_file_path, nunjucks.renderString(cpp_tmpl, idl_interface))
+    file.write(cpp_file_path, nunjucks.renderString(cpp_tmpl, idl_interface)),
+    file.write(
+        bacardi_cpp_path,
```

Comment:
Do you want to overwrite this bacardi.cc file?
I don't think this is a right way. You should write the file only one time without overwriting.